server/schedule/checker_controller.go

```diff
@@ -70,11 +73,26 @@ func (c *CheckerController) CheckRegion(region *core.RegionInfo) []*operator.Ope
 
 	if c.opts.IsPlacementRulesEnabled() {
 		if op := c.ruleChecker.Check(region); op != nil {
+			added := false
+			// ignore spilt or merge
+			if op.Kind()&operator.OpReplica != 0 {
+				missPeers, count := c.GetRuleChecker().GetMissPeer(region)
+				// miss replicate more than majority should has high priority.
+				// if less than majority, it will add waiting
+				if missPeers > count/2 {
+					c.missRegionQueue.Push(missPeers, region.GetID())
+					added = true
+				} else {
+					c.RemoveMissPeer([]uint64{region.GetID()})
+				}
+			}
 			if opController.OperatorCount(operator.OpReplica) < c.opts.GetReplicaScheduleLimit() {
 				return []*operator.Operator{op}
 			}
 			operator.OperatorLimitCounter.WithLabelValues(c.ruleChecker.GetType(), operator.OpReplica.String()).Inc()
-			c.regionWaitingList.Put(region.GetID(), nil)
+			if !added {
+				c.regionWaitingList.Put(region.GetID(), nil)
+			}
 		}
 	} else {
 		if op := c.learnerChecker.Check(region); op != nil {
```

Comment:
also need for replicaChecker, in 4.0 `PlacementRules` is not enabled in default.

Reply:
yes,it will apply to replica checker